lib/ex_force.ex

```diff
@@ -187,7 +187,8 @@ defmodule ExForce do
 
   defp do_get_sobject(client, path, fields \\ []) do
     case request(client, method: :get, url: path, query: build_fields_query(fields)) do
-      {:ok, %Tesla.Env{status: 200, body: body}} -> {:ok, SObject.build(body)}
+      {:ok, %Tesla.Env{status: 200, body: %{"attributes" => _} = body}} -> {:ok, SObject.build(body)}
+      {:ok, %Tesla.Env{status: 200, body: %{"records" => _} = body}} -> {:ok, build_result_set(body)}
```

Comment:
It's adding another possible return type to other functions calling this `do_get_sobject` func (e.g. `get_sobject`). Also the func name is "get sobject" but now it returns query result.

What about changing just `get_sobject_by_relationship` function by inlining these lines?

Reply:
Yes, that's much better, how about this?

Reply:
Changes in `get_sobject_by_relationship` looks good - then let's drop the change in `do_get_sobject`

Reply:
Oh woops, sorry! Done! Thanks for your patience 😄